expyfun/stimuli/_tracker.py

```diff
@@ -239,6 +240,20 @@ def respond(self, correct):
                 if self._direction <= 0:
                     self._direction = 1
 
+        if self._repeat_limit == 'reversals':
+            if self._x[-1] == self._x_min:
+                step_dir = 1
+                self._n_up = 0
+                reversal = True
+                self._n_reversals += 1
+                self._direction = 1
+            if self._x[-1] == self._x_max:
+                step_dir = -1
+                self._n_up = 0
+                reversal = True
+                self._n_reversals += 1
+                self._direction = 1
+
```

Comment:
They serve different purposes. A warning notifies of a potential problem (a property can't do this), and the property allows for investigating whether the potential problem is a real problem (a warning can't do this)